src/Forms/FormMessage.php

```diff
@@ -92,6 +91,57 @@ public function setMessage(
         return $this;
     }
 
+    /**
+     * Appends a message to the existing message if the types match, overrides if not
+     *
+     * @param string $message Message string
+     * @param string $messageType Message type
+     * @param string $messageCast
+     * @param bool $force Whether or not to override the existing message if appending is invalid
+     * @throws InvalidArgumentException
+     * @return $this
+     */
+    public function appendMessage(
+        string $message,
+        string $messageType = ValidationResult::TYPE_ERROR,
+        string $messageCast = ValidationResult::CAST_TEXT,
+        bool $force = false,
+    ): static {
+        if (empty($this->message)) {
+            return $this->setMessage($message, $messageType, $messageCast);
+        }
+        $canBeMerged = ($messageType === $this->getMessageType() && $messageCast === $this->getMessageCast());
+
+        if (!$canBeMerged && $force === false) {
+            throw new InvalidArgumentException(
+                sprintf(
+                    "Couldn't append message of type %s and cast %s to existing message of type %s and cast %s",
+                    $messageType,
+                    $messageCast,
+                    $this->getMessageType(),
+                    $this->getMessageCast(),
+                )
+            );
+        }
+
+        // Checks that the exact message string is not already contained before appending
+        $messageContainsString = strpos($this->message, $message) !== false;
+        if ($canBeMerged && $messageContainsString) {
+            return $this;
+        }
+
+        if ($canBeMerged) {
+            $message = sprintf(
+                '%s%s%s',
+                $this->message,
+                $messageCast === ValidationResult::CAST_HTML ? '<br />' : PHP_EOL,
+                $message
+            );
```

Comment:
I think concatenation or implode are more readable in this scenario

**Concatenation:**
```suggestion
            $separator = $messageCast === ValidationResult::CAST_HTML ? '<br />' : PHP_EOL;
            $message = $this->message . $separator . $message;
```

**Implode:**
```suggestion
            $separator = $messageCast === ValidationResult::CAST_HTML ? '<br />' : PHP_EOL;
            $message = implode($separator, [$this->message, $message]);
```

Reply:
I chooooooooooose concat 😄